### PR TITLE
[SPARK-42570][CONNECT][PYTHON] Fix DataFrameReader to use the default source

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Fixes `DataFrameReader` to use the default source.

### Why are the changes needed?

```py
spark.read.load(path)
```

should work and use the default source without specifying the format.

### Does this PR introduce _any_ user-facing change?

The `format` doesn't need to be specified.

### How was this patch tested?

Enabled related tests.

### CODE REVIEW

File: python/pyspark/sql/tests/test_readwriter.py [hunk 1]
```diff
@@ -31,75 +31,77 @@ def test_save_and_load(self):
...
-        self.assertEqual(sorted(df.collect()), sorted(actual.collect()))
-
-        defaultDataSourceName = self.spark.conf.get(
-            "spark.sql.sources.default", "org.apache.spark.sql.parquet"
-        )
-        self.spark.sql("SET spark.sql.sources.default=org.apache.spark.sql.json")
-        actual = self.spark.read.load(path=tmpPath)
-        self.assertEqual(sorted(df.collect()), sorted(actual.collect()))
-        self.spark.sql("SET spark.sql.sources.default=" + defaultDataSourceName)
+        try:
```
Comment (ueshin) [inline python/pyspark/sql/tests/test_readwriter.py:34]: This change is to make the cleanup done properly.

**Review by HyukjinKwon — APPROVED** [review]
Nice, thanks Takuya

**Comment by amaliujia** [comment]
LGTM

What is the default source BTW?

File: connector/connect/common/src/main/protobuf/spark/connect/relations.proto [hunk 2]
```diff
@@ -122,8 +122,8 @@ message Read {
   }
 
   message DataSource {
-    // (Required) Supported formats include: parquet, orc, text, json, parquet, csv, avro.
-    string format = 1;
+    // (Optional) Supported formats include: parquet, orc, text, json, parquet, csv, avro.
+    optional string format = 1;
```
Comment (amaliujia) [inline connector/connect/common/src/main/protobuf/spark/connect/relations.proto:126]: Nit: can you improve the documentation here to see what is the default format and/or how to configure it if there is a way?

**Comment by ueshin** [comment]
> What is the default source BTW?

If format is not set, the value from SQL conf 'spark.sql.sources.default' will be used.

**Comment by hvanhovell** [comment]
Merging.

**Pull request closed without merging**